jarviscli/plugins/gmail.py

```diff
@@ -1,47 +1,163 @@
-    '''
     try:
-        # establshing server connection
-        server = smtplib.SMTP("smtp.gmail.com", 587)
-        server.ehlo()
-        server.starttls()
-        print("SERVER CONNECTED")
+        smtp = smtplib.SMTP(EMAIL_HOST, EMAIL_PORT)
+        smtp.starttls()
     except BaseException:
-        # in case of failure
         print("Could Not connect to Gmail")
         return
-    user = jarvis.input("User id: ")                          # YOUR ID
-    Pass_w = jarvis.input("\nPassword: ", password=True)                # YOUR Password
 
     try:
-        server.login(user, Pass_w)                           # user log in
-        print("User Logged in")
+        smtp.login(email_user, email_pass)
+        smtp.send_message(final_message)
+        print("Sent!")  # confirmation
+    except Exception:
+        print("Unable to send text right now, please try again later.")
+    finally:
+        smtp.close()
+
+
+def send_mail(send_to, send_from, final_message, email_user, email_pass):
+    """Send email to recipient."""
+
+    try:
+        smtp = smtplib.SMTP(EMAIL_HOST, EMAIL_PORT)
+        smtp.starttls()
     except BaseException:
-        print(
-            '''Allow Less secure apps in GOOGLE ACCOUNT SETTINGS to use SMTP services by following the given steps:
-                                                                      \n\t\tStep 1. Log in to email using your browser.
-                                                                      \n\t\tStep 2. Go to account settings.
-                                                                      \n\t\tStep 3. Find 'allow less secure apps' and mark it as ON.''')
-        server.quit()
+        print("Could Not connect to Gmail")
         return
 
-    receiver_id = jarvis.input("\nEnter receiver id\n")             # Reciever ID
-    msg = jarvis.input("\nEnter message\n")                         # message
-    server.sendmail(user, receiver_id, msg)
+    try:
+        smtp.login(email_user, email_pass)
+        smtp.sendmail(send_from, send_to, final_message)
+        print("Sent!")  # confirmation
+    except Exception as e:
+        print("Unable to send text right now, please try again later.")
+        print(e.with_traceback())
+    finally:
+        smtp.close()
+
+
+@plugin('gmail')  # decorator
+def gmail(jarvis, s):
+    """
+    Sending email from a gmail account using SMTP services.
+    To use this plugin :
+                1. User should have a gmail id.
+                2. Less secure apps should be allowed to access the gmail account.
+    """
+    user, pass_word = get_user_pass(jarvis)
+    receiver_id = jarvis.input("\nEnter receiver id\n")  # Reciever ID
+    msg = jarvis.input("\nEnter message\n")  # message
+    send_mail(receiver_id, user, msg, user, pass_word)
+
+
+@require(network=True, platform=[Platform.MACOS, Platform.LINUX])
+@alias("message")
+@plugin("text")
+def text(jarvis, s):
```

Comment:
Fascinating. Didn't even realize there are services to just send a mail to a phone number ;)

Reply:
😄